src/execution_plan/ops/shared/create_functions.c

```diff
@@ -157,19 +157,45 @@ void CommitNewEntities(OpBase *op, PendingCreations *pending) {
 }
 
 // Resolve the properties specified in the query into constant values.
-PendingProperties *ConvertPropertyMap(Record r, const PropertyMap *map) {
+PendingProperties *ConvertPropertyMap(Record r, PropertyMap *map, bool fail_on_null) {
 	PendingProperties *converted = rm_malloc(sizeof(PendingProperties));
-	converted->property_count = map->property_count;
-	converted->attr_keys = map->keys; // This pointer can be copied directly.
 	converted->values = rm_malloc(sizeof(SIValue) * map->property_count);
 	for(int i = 0; i < map->property_count; i++) {
-		converted->values[i] = AR_EXP_Evaluate(map->values[i], r);
-		// Emit an error and return NULL if we're trying to add an invalid type.
-		if(!(SI_TYPE(converted->values[i]) & SI_VALID_PROPERTY_VALUE)) {
-			Error_InvalidPropertyValue();
-			break;
+		SIValue val = AR_EXP_Evaluate(map->values[i], r);
+		if(!(SI_TYPE(val) & SI_VALID_PROPERTY_VALUE)) {
+			// This value is of an invalid type.
+			if(!SIValue_IsNull(val)) {
+				// If the value was a complex type, emit an exception.
+				converted->property_count = i;
+				PendingPropertiesFree(converted);
+				Error_InvalidPropertyValue();
+				ErrorCtx_RaiseRuntimeException(NULL);
+			}
+			/* The value was NULL. If this was prohibited in this context, raise an exception,
+			 * otherwise skip this value. */
+			if(fail_on_null) {
+				// Emit an error and exit.
+				converted->property_count = i;
+				PendingPropertiesFree(converted);
+				ErrorCtx_RaiseRuntimeException("Cannot merge node using null property value");
+			} else {
+				// Replace the NULL value with the final conversion and reset the loop.
+				Attribute_ID *keys = (Attribute_ID *)map->keys;
+				AR_EXP_Free(map->values[i]);
```

Comment:
`CREATE (), ( {v:1})`
`MATCH (n) CREATE ({x:n.v})`

Assuming the fist `n` node resolves to the empty node, as such `n.v` is NULL, which will remove the expressions `n.v` from the shared property map, causing the second created node to miss its `x` attribute.